src/reducers/LeaderboardReducer.js

```diff
@@ -0,0 +1,35 @@
+import _ from 'lodash';
+
+import {
+  FETCH_LEADERBOARD
+} from '../actions/types';
+
+const initialState = {
+  users: []
+};
+
+const LeaderboardReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case FETCH_LEADERBOARD:
+      if(_.isEqual(state.users, action.payload)) {
+        return {
+          ...state,
+        }
+      }
+      else {
+        // Sort here to improve efficiency. Really, we shouldn't be doing this
+        // array comparison at all though.
+        // TODO: Look into Immutables.
```

Comment:
We should use [OrderedSet](https://immutable-js.github.io/immutable-js/docs/#/OrderedSet), and look into doing this when we convert to TypeScript #32